tablets_split_merge_test.py

```diff
@@ -0,0 +1,163 @@
+        with self.db_cluster.cql_connection_patient(self.db_cluster.nodes[0]) as session:
+            session.execute(create_scylla_bench_table_query())
+
+    def _start_stress_if_not_running(self, stress_queue, stress_params):
+        if self._is_stress_finished(stress_queue=stress_queue):
+            InfoEvent(message="The Stress is not running. Rerunning it.").publish()
+            self._run_all_stress_cmds(stress_queue, stress_params)
+
+    def test_tablets_split_merge(self):  # pylint: disable=too-many-locals  # noqa: PLR0914
+        """
+        (1) writing initial schema and get initial number of tablets.
+        (2) start write and read stress simultaneously.
+        (3) wait for tablets split.
+        (4) wait for stress completion.
+        (5) run deletions, nodetool flush and major compaction.
+        (6) wait for tablets merge, following a shrunk dataset size.
+        (7) redo more such cycles.
+        """
+
+        # (1) initiate schema.
+        InfoEvent(message=f"Create c-s keyspace with tablets initial value").publish()
+        self._pre_create_large_partitions_schema()
+        stress_cmd = self.params.get('stress_cmd')
+        stress_read_cmd = self.params.get('stress_read_cmd')
+        deletion_percentage = 80  # How many of dataset partitions should be deleted.
+
+        # Run prepare stress
+        self.run_prepare_write_cmd()
+
+        # Run Read background stress
+        stress_read_queue = []
+        read_params = {'keyspace_num': 1, 'stress_cmd': stress_read_cmd}
+        self._run_all_stress_cmds(stress_read_queue, read_params)
+        self.node1 = self.db_cluster.nodes[0]
+
+        # Run cycles of writes and deletions
+        cycles_num = 10
+        for cycle in range(cycles_num):
+            tablets_num = self._get_tablets_number()
+            InfoEvent(message=f"Cycle {cycle }, Initial tablets number before stress is: {tablets_num}").publish()
+            InfoEvent(message=f"Starting write load: {stress_cmd}").publish()
+            stress_queue = []
+            self.assemble_and_run_all_stress_cmd(stress_queue=stress_queue, stress_cmd=stress_cmd, keyspace_num=1)
+
+            self._start_stress_if_not_running(stress_read_queue, read_params)
+
+            InfoEvent(message="Wait for write stress to finish (if running)").publish()
+            for stress in stress_queue:
+                self.verify_stress_thread(cs_thread_pool=stress)
+
+            InfoEvent(message="Wait for tablets split following writes.").publish()
+            self._wait_for_tablet_split(tablets_num)
+
+            tablets_num = self._get_tablets_number()
+            self._start_stress_if_not_running(stress_read_queue, read_params)
+            InfoEvent(message=f"Start deletions to trigger a tablets merge following {tablets_num} tablets.").publish()
+            self.delete_partitions(deletion_percentage)
+            tablets_num_post_deletions = self._get_tablets_number()
```

Comment:
the test gets it as well. this one expects a maximum tablets number before flush and compact.
actually i'm if flush and compaction have or should have any impact on tablets since it's only indirectly impact it.
perhaps @raphaelsc can advise whether a major compaction can directly trigger a tablet merge or the dataset size is calculated differently for this trigger.